stellar-dotnet-sdk-test/RevokeSponsorshipTest.cs

```diff
@@ -0,0 +1,441 @@
+
+namespace stellar_dotnet_sdk_test;
+
+[TestClass]
+public class RevokeSponsorshipTest
+{
+    private readonly Server _server = new("https://horizon-testnet.stellar.org");
+    private string _sponsoredId = "GC3TDMFTMYZY2G4C77AKAVC3BR4KL6WMQ6K2MHISKDH2OHRFS7CVVEAF";
+    private string _sponsoringId = "GARRDNS77ZSI6PPXRBWTHIVX4RS2ULVBKNJXFRV77AZUNLDUNV2NAHJA";
+
+    private Asset _assetA =
+        new AssetTypeCreditAlphaNum4("AAA", "GARRDNS77ZSI6PPXRBWTHIVX4RS2ULVBKNJXFRV77AZUNLDUNV2NAHJA");
+
+    private KeyPair _sponsoringAccount =
+        KeyPair.FromSecretSeed("SDR4PTKMR5TAQQCL3RI2MLXXSXQDIR7DCAONQNQP6UCDZCD4OVRWXUHI");
+
+    private KeyPair _sponsoredAccount =
+        KeyPair.FromSecretSeed("SDBNUIC2JMIYKGLJUFI743AQDWPBOWKG42GADHEY3FQDTQLJADYPQZTP");
+
+    private const string DataName = "my secret";
+
+    [TestInitialize]
+    public async Task Setup()
+    {
+        Network.UseTestNetwork();
+        // Generates a new _accountId in case the testnet has been reset
+        try
+        {
+            await _server.Accounts.Account(_sponsoringId);
+        }
+        catch (AccountNotFoundException)
+        {
+            _sponsoringAccount = await CreateNewRandomAccountOnTestnet();
+        }
+
+        try
+        {
+            await _server.Accounts.Account(_sponsoredId);
+        }
+        catch (AccountNotFoundException)
+        {
+            _sponsoredAccount = await CreateNewRandomAccountOnTestnet();
+        }
+
+        _sponsoringId = _sponsoringAccount.AccountId;
+        _sponsoredId = _sponsoredAccount.AccountId;
+        _assetA = new AssetTypeCreditAlphaNum4("AAA", _sponsoringId);
+    }
+
+    [TestCleanup]
+    public void Cleanup()
+    {
+        Network.Use(null);
+        _server.Dispose();
+    }
+
+    private async Task<KeyPair> CreateNewRandomAccountOnTestnet()
+    {
+        bool isSuccess;
+        KeyPair account;
+        do
+        {
+            account = KeyPair.Random();
+            var fundResponse = await _server.TestNetFriendBot.FundAccount(account.AccountId).Execute();
+            var result = TransactionResult.FromXdrBase64(fundResponse.ResultXdr);
+            isSuccess = result.IsSuccess && result is TransactionResultSuccess;
+        } while (!isSuccess);
+
+        return account;
+    }
+
+    private async Task RevokeClaimableBalanceSponsorship(string balanceId)
+    {
+        Assert.IsNotNull(balanceId);
+        var account = await _server.Accounts.Account(_sponsoringId);
+
+        var revokeOperation = new RevokeLedgerEntrySponsorshipOperation.Builder(balanceId).Build();
+
+        var beginSponsoringOperation = new BeginSponsoringFutureReservesOperation.Builder(_sponsoringId)
+            .SetSourceAccount(_sponsoredAccount)
+            .Build();
+        var endSponsoringOperation = new EndSponsoringFutureReservesOperation.Builder().Build();
+        
+        var tx = new TransactionBuilder(account)
+            .AddOperation(beginSponsoringOperation)
```

Comment:
Do we need to have a begin and end sponsorship if we are revoking a sponsorship? That doesn't make sense to me.

Reply:
In this test, we are transferring the sponsorship. I don't know why, but it didn't allow me to just revoke it. This was the only option that worked for a claimable balance revocation.